lib/roby/interface/client.rb

```diff
@@ -336,8 +337,21 @@ def async_call(path, m, *args, &block)
                     end
                 end
                 io.write_packet([path, m, *args])
-                pending_async_calls << { block: block, path: path,
-                                         m: m, args: args }
+                pending_async_calls << { block: block, path: path, m: m, args: args }
+                pending_async_calls.last
```

Comment:
I'm OK-ish to use the hash as identifier, but then I would freeze to at least ensure that the caller can't mess with it. An alternative would be to use a plain object (`Object.new`) as a new `id` element in the hash, and return that.